packages/browser-sdk/src/client.ts

```diff
@@ -190,6 +190,83 @@ export class BucketClient {
     }
   }
 
+  /**
+   * Update the user context.
+   * @description Performs a shallow merge with the existing user context.
+   * Updates to the user ID will be ignored.
+   *
+   * @param user
+   */
+  async updateUser(user: { [key: string]: string | number | undefined }) {
+    if (user.id && user.id !== this.context.user?.id) {
+      this.logger.warn(
+        "ignoring attempt to update the user ID. Re-initialize the BucketClient with a new user ID instead.",
+      );
+      return;
+    }
+
+    this.context.user = {
+      ...this.context.user,
+      ...user,
+      id: user.id ?? this.context.user?.id,
+    };
+    void this.user();
+    await this.featuresClient.setContext(this.context);
+  }
+
+  /**
+   * Update the company context.
+   * Performs a shallow merge with the existing company context.
+   * Updates to the company ID will be ignored.
+   *
+   * @param company
+   */
+  async updateCompany(company: { [key: string]: string | number | undefined }) {
+    if (company.id && company.id !== this.context.company?.id) {
```

Comment:
This check would refuse to update the company if the existing context doesn't have a company set, since the new company ID would be defined and the existing one would be `undefined`.

Reply:
It's confusing because there's `id: company.id ?? this.context.company?.id` on line `234` which made me think that is an option. Maybe also update the comment to say "attempts to update the company ID will result in the call being discarded"? Otherwise it sounds like the company ID is ignored, but not the other parts.

Reply:
gotcha, makes sense